src/fastify-plugins/maps/index.js

```diff
@@ -0,0 +1,24 @@
+import fp from 'fastify-plugin'
+
+import { plugin as StaticMapsPlugin } from './static-maps.js'
+
+export const PLUGIN_NAME = 'mapeo-maps'
+
+export const plugin = fp(mapsPlugin, {
+  fastify: '4.x',
+  name: PLUGIN_NAME,
+})
+
+/**
+ * @typedef {{ prefix?: string } & Partial<import('./static-maps.js').StaticMapsPluginOpts>} MapsPluginOpts
+ */
+
+/** @type {import('fastify').FastifyPluginAsync<MapsPluginOpts>} */
+async function mapsPlugin(fastify, opts) {
+  if (opts.staticRootDir) {
+    fastify.register(StaticMapsPlugin, {
+      prefix: opts.prefix,
+      staticRootDir: opts.staticRootDir,
+    })
+  }
```

Comment:
wondering if i should just avoid conditionally registering here for the sake of simplifying the implementation a bit. not sure if there's a case where the end user wouldn't want to support the static map serving functionality